workspace/em.core/em.utils/AlarmMgr.em.zig

```diff
@@ -68,13 +69,12 @@ pub const EM__TARG = struct {
 
```

Comment:
lacking any other testing, let's assume this is "correct"....

but now, ask this question:  why isn't the comparison simply an **equals** test on `_thresh`???

when `cur_alarm` was set, we used a thresh value computed at that time; and indeed, there could have been more than one alarm that would ring at the same (future) time....

the fact that we **enter** the wakeup handler at all means that we have crossed the thresh value of `cur_alarm`; simply find that alarm (and others with the same value is probably sufficient)!!!

assuming the calculation of `ticksToThresh` is correct (modulo wrapping), this might be sufficient....

Reply:
I think this only works if it's guaranteed that you come into the handler at exactly cur.thresh == a.thresh.  The logic that is there handles that it might get there after the fact too (i.e. cur.thresh is at least a.thresh).  If you're guaranteeing that we always get there at the exact right moment, then we could simply test for a.thresh == cur.thresh.  But I think this is more robust.